lib/phoenix_live_view/helpers.ex

```diff
@@ -478,6 +482,99 @@ defmodule Phoenix.LiveView.Helpers do
     EEx.compile_string(expr, options)
   end
 
+  @doc """
+  Provides `~H` sigil with HTML safe Live `HEEx` syntax inside source files.
+
+  > Note: `HEEx` requires Elixir >= `1.12.0` in order to provide accurate file:line:column information
+  > in error messages. Earlier Elixir versions will work but will show inaccurate error messages.
+
+  `HEEx` is a HTML-aware and component-friendly extension of `EEx` that provides:
+
+    * Built-in handling of HTML attributes
+    * An HTML-like notation for injecting function components
+    * Compile-time validation of the structure of the template
+
+  ## Example
+
+      def render(assigns) do
+        ~H"\""
+        <div title="My div" class={@class}>
+          <MyApp.Weather.component city="Kraków"/>
+        </div>
+        "\""
```

Comment:
@msaraiva I have a question but it's purely in relation to aesthetics, so feel free to dismiss it :)

Is it possible / would it be worth the additional complexity to allow invoking a component with only its module name?

```ex
<MyApp.Weather, city="Kraków"/>
```

I assume that if it had made sense to do so you would have done it that way, again I am just curious :)

Also are the escaped quotes necessary, here? I do not think I have seen this example before.

Reply:
We decided to make an explicit distinction between functional components, that are stateless, and stateful components, which are modules. So we may add affordances later, but right now we want to be explicit on their differences.

> Also are the escaped quotes necessary, here? I do not think I have seen this example before.

It is because we are inside docs.

Reply:
Thanks @josevalim that all makes sense

> It is because we are inside docs.

I have seen various incantations of sigils and quotes escaping the _whole_ doc block– this inline version was new to me and I like it :)